lib/manageiq_performance/middleware.rb

```diff
@@ -90,15 +90,30 @@ def save_report env, prefix, long_form, short_form
     end
   end
 
-  def self.profile name = nil, &code
-    name ||= caller.first.match(/`(|.*\s)([a-z_\(\)]+)>?'$/)[2].gsub(/[^a-z_]/,'')
+  def self.profile *args, &code
+    options = args.delete_at(args.find_index { |arg| arg.is_a? Hash } || args.size + 1) || {}
```

Comment:
Yeah, I think I saw a better way to do this in the bundler source, but I forgot my rational for this, so let me look into that.